src/libsync/syncengine.cpp

```diff
@@ -1473,7 +1465,18 @@ QByteArray SyncEngine::getPermissions(const QString &file) const
             return rx.cap(1).toLatin1();
         }
     }
-    return _remotePerms.value(file);
+
+    // Fetch from the csync context while we still have it.
+    ASSERT(_csync_ctx->status != CSYNC_STATUS_INIT);
+
+    if (file == QLatin1String(""))
+        return _csync_ctx->remote.root_perms;
+
+    auto it = _csync_ctx->remote.files.find(file.toUtf8());
```

Comment:
too bad we now have to allocate only to convert to a QByteArray,  but i guess this is compensated by the fact we did not insert it into the map in the first place.